Framework/DataHandling/src/LoadEventNexus.cpp

```diff
@@ -677,9 +681,12 @@ std::shared_ptr<BankPulseTimes> LoadEventNexus::runLoadNexusLogs(
     // Get the period log. Map of DateAndTime to Period int values.
     if (run.hasProperty("period_log")) {
       auto *temp = run.getProperty("period_log");
+      std::string status = "";
       // Check for corrupted period logs
       std::unique_ptr<TimeSeriesProperty<int>> tempPeriodLog(dynamic_cast<TimeSeriesProperty<int> *>(temp->clone()));
-      checkForCorruptedPeriods(std::move(tempPeriodLog), periodLog, nPeriods, nexusfilename);
+      nPeriods = checkForCorruptedPeriods(std::move(tempPeriodLog), periodLog, nPeriods, nexusfilename, status);
+      if (!status.empty())
+        alg.getLogger().warning(status);
```

Comment:
Why not send the logger warning directly from `checkForCorruptedPeriods`?

Reply:
`checkForCorruptedPeriods` is static so it doesn't know about the logger, unless it got passed as an argument, could do that instead.

Reply:
ah I see, no problem to leave it as is